In [10]:
# magic function para hacer que los graficos de matplotlib se renderizen en el notebook.
%matplotlib notebook

import datetime as datetime
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

plt.style.use('default') # Make the graphs a bit prettier
plt.rcParams['figure.figsize'] = (15, 5)

## Carga de datos y visualización de cada csv

In [11]:
educacion = pd.read_csv('fiuba_1_postulantes_educacion.csv',sep=',', error_bad_lines=False, encoding='ISO-8859-1')
educacion.tail()

idpostulante              nombre      estado
298226      GNZqARp  Terciario/Técnico  Abandonado
298227      ow2p8vk       Universitario  Abandonado
298228      6rPEvwv       Universitario  Abandonado
298229      vVja45X       Universitario  Abandonado
298230      1QPMbX3       Universitario  Abandonado

In [12]:
genero_y_edad = pd.read_csv('fiuba_2_postulantes_genero_y_edad.csv', sep=',', error_bad_lines=False, encoding='ISO-8859-1')
genero_y_edad.tail()

idpostulante fechanacimiento        sexo
200883      1QPMbX3      1988-01-07        MASC
200884      0zPa0eq      1998-05-09         FEM
200885      2zP6Jd0      1997-03-06         FEM
200886      8MPoxDW      1997-07-07        MASC
200887      2zP6J9V             NaN  NO_DECLARA

In [13]:
visitas = pd.read_csv('fiuba_3_vistas.csv', sep=',', error_bad_lines=False, encoding='ISO-8859-1')
visitas.tail()

idAviso                     timestamp idpostulante
961892  1112366944  2018-02-28T23:59:08.772-0500      MVPj3AK
961893  1001442844  2018-02-28T23:59:37.000-0500      VNrYP6m
961894  1112362644  2018-02-28T23:59:46.181-0500      owzLD43
961895  1112305277  2018-02-28T23:59:52.754-0500      A3DraY1
961896  1112345900  2018-02-28T23:59:55.409-0500      ekmNzBW

In [14]:
postulaciones = pd.read_csv('fiuba_4_postulaciones.csv', sep=',', error_bad_lines=False, encoding='ISO-8859-1')
postulaciones.tail()

idaviso idpostulante     fechapostulacion
3401618  1111753681      0zPa0eq  2018-02-28 23:54:44
3401619  1112369306      2zP6Jd0  2018-02-28 23:57:43
3401620  1112308651      8MPoxDW  2018-02-28 23:52:13
3401621  1112341128      8MPoxDW  2018-02-28 23:59:06
3401622  1112369825      2zP6J9V  2018-02-28 23:56:10

In [15]:
avisos_online = pd.read_csv('fiuba_5_avisos_online.csv', sep=',', error_bad_lines=False, encoding='ISO-8859-1')
avisos_online.tail()

idaviso
5023  1112387966
5024  1112383275
5025  1112388199
5026  1112388376
5027  1112387833

In [16]:
avisos_detalle = pd.read_csv('fiuba_6_avisos_detalle.csv', sep=',', error_bad_lines=False, encoding='ISO-8859-1')
avisos_detalle.tail()

idaviso  idpais                                             titulo  \
13529  1112342505       1                      Vendedor de Salon -- UNOMOTOS   
13530  1112342513       1                OPERADOR DE ENSAYOS NO DESTRUCTIVOS   
13531  1112342456       1  Back officer Junior de Fondos Comunes de Inver...   
13532  1112342496       1                           Asistente Administrativo   
13533  1112342432       1  Ingeniero para coordinación de obras de canal...   

                                             descripcion        nombre_zona  \
13529  <p>UNOMOTOS Concesionaria líder en el rubro d...  Gran Buenos Aires   
13530  <p><strong>RECONOCIDA EMPRESA DE MONTAJES INDU...  Gran Buenos Aires   
13531  <p>BNP Paribas Investment Partners, importante...  Gran Buenos Aires   
13532  <p>Somos un joven grupo gastronómico, que se ...  Gran Buenos Aires   
13533  <p>Importante empresa de servicios para obras ...  Gran Buenos Aires   

      ciudad mapacalle tipo_de_trabajo                          nivel_laboral  \
13529    NaN       NaN       Full-time                   Senior / Semi-Senior   
13530    NaN       NaN       Full-time                   Senior / Semi-Senior   
13531    NaN       NaN       Full-time                                 Junior   
13532    NaN       NaN       Full-time  Gerencia / Alta Gerencia / Dirección   
13533    NaN       NaN       Full-time        Jefe / Supervisor / Responsable   

                                   nombre_area  \
13529                                   Ventas   
13530  Ingeniería Oficina Técnica / Proyecto   
13531                             Contabilidad   
13532                          Administración   
13533                            Construcción   

                                denominacion_empresa  
13529                                       UNOMOTOS  
13530                                Grupo Consultor  
13531  BNP PARIBAS ARGENTINA INVESTMENT PARTNERS S A  
13532                             Cerveteca Odin Srl  
13533                                      CIMA S.A.

## Merge entre los archivos educación y género y edad de los postulantes

Unimos los datos de ambos data frame con el fin de colocar todos los datos de los postulantes en un mismo data frame

In [17]:
postulantes = pd.merge(educacion, genero_y_edad, on='idpostulante', how='outer')
postulantes.tail()

idpostulante nombre estado fechanacimiento        sexo
310362      mzdNo99    NaN    NaN      1999-08-10         FEM
310363      Nzr1J34    NaN    NaN             NaN  NO_DECLARA
310364      JBrNMNk    NaN    NaN      1997-01-11        MASC
310365      2zP6Jd0    NaN    NaN      1997-03-06         FEM
310366      2zP6J9V    NaN    NaN             NaN  NO_DECLARA

## Merge entre los archivos avisos_online y avisos_detalle

Unimos ambos datos frame con el fin de colocar los datos de todos los avisos en un mismo data frame

In [18]:
avisos = pd.merge(avisos_online, avisos_detalle, on='idaviso', how='outer')
avisos.tail()

idaviso  idpais                                             titulo  \
13844  1112342204     1.0  Ejecutivo de Ventas - Prov. de Bs. As e Interi...   
13845  1112341862     1.0  24 Feb - Chief Software Architect ($100K/yr) -...   
13846  1112341899     1.0  24 Feb - Customer Support Manager ($100K/yr) -...   
13847  1112341992     1.0  24 Feb - VP of Customer Support ($200K/yr) - O...   
13848  1112342193     1.0                        Supervisor de Mantenimiento   

                                             descripcion        nombre_zona  \
13844  <p>Editorial Rubinzal Culzoni, líder en el á...  Gran Buenos Aires   
13845  <p>Para inscribirse en el torneo debe ingresar...  Gran Buenos Aires   
13846  <p>Para inscribirse en el torneo debe ingresar...  Gran Buenos Aires   
13847  <p>Para inscribirse en el torneo debe ingresar...  Gran Buenos Aires   
13848  <p>Nuestro cliente, un importante PYME dedicad...  Gran Buenos Aires   

      ciudad mapacalle tipo_de_trabajo                    nivel_laboral  \
13844    NaN       NaN       Full-time             Senior / Semi-Senior   
13845    NaN       NaN     Teletrabajo  Jefe / Supervisor / Responsable   
13846    NaN       NaN     Teletrabajo  Jefe / Supervisor / Responsable   
13847    NaN       NaN     Teletrabajo  Jefe / Supervisor / Responsable   
13848    NaN       NaN       Full-time  Jefe / Supervisor / Responsable   

                 nombre_area denominacion_empresa  
13844                 Ventas   Rubinzal Editorial  
13845  Tecnologia / Sistemas            CrossOver  
13846       Soporte Técnico            CrossOver  
13847   Atención al Cliente            CrossOver  
13848          Mantenimiento        VF CONSULTING

## Analizando el set de datos de avisos

Se analiza este set de datos puntualmente, ya que queremos filtrar datos que pertenezcan a CABA y GBA.
En particular haremos enfazis en las columnas, idpais, nombre_zona, ciudad y mapacalle 

In [11]:
avisos.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 13849 entries, 0 to 13848
Data columns (total 11 columns):
idaviso                 13849 non-null int64
idpais                  13534 non-null float64
titulo                  13534 non-null object
descripcion             13534 non-null object
nombre_zona             13534 non-null object
ciudad                  47 non-null object
mapacalle               872 non-null object
tipo_de_trabajo         13534 non-null object
nivel_laboral           13534 non-null object
nombre_area             13534 non-null object
denominacion_empresa    13529 non-null object
dtypes: float64(1), int64(1), object(9)
memory usage: 1.3+ MB


## CHIC@S
### 1) considero que no hay que tomar la col ciudad y mapacalle, ya que existe 13802 null y 12977 null, son datos 

In [13]:
avisos.isnull().sum()

idaviso                     0
idpais                    315
titulo                    315
descripcion               315
nombre_zona               315
ciudad                  13802
mapacalle               12977
tipo_de_trabajo           315
nivel_laboral             315
nombre_area               315
denominacion_empresa      320
dtype: int64

## CHIC@S
### 2) Considero en limpiar el dato Bs As (fuera de GBA) 

In [19]:
#Verificamos que las zonas sean solo de Cap Fed y GBA
avisos.nombre_zona.value_counts()

Gran Buenos Aires              12654
Capital Federal                  876
Buenos Aires (fuera de GBA)        2
GBA Oeste                          2
Name: nombre_zona, dtype: int64

## CHIC@S
### 3) Analizando que datos diferentes a null existe en la col ciudad, se puede ver que hay datos mal cargados, esto tb es otro motivo más (a mi parecer) de no considerar para nuestro analisis esta columna

In [20]:
#Verificamos que las ciudades pertenezcan sólo a CABA y GBA
avisos.ciudad.value_counts()

Buenos Aires                 14
Argentina                    13
CABA                          3
Capital Federal               2
San Isidro                    2
Santa Rosa                    1
paternal                      1
Tortuguitas                   1
Vicente Lopez                 1
caba                          1
Barracas                      1
Buenos Aires Province         1
Zárate, Campana, Escobar     1
Mendoza                       1
República Argentina          1
Microcentro                   1
La Plata                      1
Parque Patricios              1
Name: ciudad, dtype: int64

In [16]:
avisos.mapacalle.value_counts()

SARMIENTO 1937                                 43
Microcentro                                    30
Buenos Aires                                   24
Palermo                                        23
Barracas                                       18
Puerto Madero                                  15
Belgrano                                       14
Reconquista 865                                13
retiro                                         12
Recoleta                                       10
Parque Patricios                               10
nuñez                                          8
saavedra                                        8
Peron 955                                       7
Mataderos                                       6
Monserrat                                       6
San Telmo                                       6
25 de Mayo 457 7º piso                         6
Pasaje Aguado 628                               6
Amenábar 1074                                  5


## Limpieza del set de datos de avisos
Se limpia todos los nans que pueda aparecer en la columna nombre_zona, la justificación es los ptos 1) y 2) que se recalco 

In [22]:
avisos.dropna(subset=['nombre_zona'],inplace=True)

## Filtración de datos por CABA y GBA

In [27]:
avisosGran_Bs_As=avisos.loc[avisos.nombre_zona.str.contains('Gran Buenos Aires'),:]
avisosGBA_Oeste =avisos.loc[avisos.nombre_zona.str.contains('GBA Oeste'),:]

avisosGBA=pd.concat([avisosGBA_Oeste,avisosGran_Bs_As])

In [29]:
avisosCABA=avisos.loc[avisos.nombre_zona.str.contains('Capital Federal'),:]

avisosCABA_GBA=pd.concat([avisosCABA,avisosGBA])

## Eliminación de columnas
Se limpia las columnas por los siguientes motivos:

  * idpais: existe 315 datos nullos y representa menos del 2,5% .Los datos diferentes a nulos, tienen un unico id
   
  * ciudad: existe 13802 datos nulos y aquellos datos diferente a nulos, son datos mal cargados
   
  * mapacalle: existe 12977 datos nulos (representa el 93,7% del total de los datos),no es relevante para nuestro      analisis


In [30]:
avisosCABA_GBA.drop(["idpais", "ciudad", "mapacalle"],axis=1,inplace=True)


In [31]:
avisosCABA_GBA

idaviso                                             titulo  \
135    1112302920                         Ejecutivo Comercial Retail   
143    1112326553                       Ejecutivo Comercial AudioCar   
271    1111840946                          Recruiter IT Jr Bilingüe   
366    1112366334                                        Data Entry    
495    1112362278                       Repositores Externos / LUJAN   
521    1110661032  Operador 2º nivel - Operaciones y Monitoreo (...   
522    1111102056                    Ejecutivo de Venta Telefónica    
523    1111105892  Ejecutivo/a de Ventas para segmento Residencia...   
524    1112221849                                 Relevador de campo   
525    1112322267                              Pasante de Tesorería   
526    1112358985  Representante de Atención, Retención y Fidel...   
528    1112371120          Asistente para Generación de Entrevistas   
615    1112347309  ¡MUSIMUNDO busca personal para las áreas de ...   
619    1112359554  ¡MUSIMUNDO busca personal para linea de CAJA ...   
659    1112336556                            Jóvenes Profesionales    
670    1112374787  Supervisor de Mantenimiento Flota - Villa Soldati   
679    1112149713                       Facility Manager Coordinator   
817    1112337831                   Promovendedor (San Luis Capital)   
818    1112337843                       Repositor (San Luis Capital)   
903    1112322345       Operario de Deposito con Manejo de Apiladora   
929    1112353391                          Administrativo Contable     
1030   1112377974                          Asesor Comercial Nordelta   
1037   1112368503                   Desarrollador Full Stack SSr.-Sr   
1048   1112239168      Asistente de Impuestos Jr. - Inglés Avanzado   
1196   1112364857  Asistente Comercial de Seguros de Automotor - ...   
1207   1112322859  Distribuidores Domiciliarios ( Carteros), con ...   
1209   1112358477  Especialista e-commerce: certificacion Google,...   
1210   1112358897  Gerente de Mantenimiento y afines - Zona Capit...   
1211   1112358949                 Liquidador de Sueldo - Zona: Pilar   
1220   1112053606  Profesionales Graduados en Ciencias de la Salu...   
...           ...                                                ...   
13819  1112341225  Feb 24 - IT Project Manager - Online Hiring To...   
13820  1112340837                         Costureras con experiencia   
13821  1112341074                                           OPERARIO   
13822  1112341257  Analista Contable - Experiencia en Recursos Hu...   
13823  1112341393                       Comercial Agencia de Turismo   
13824  1112341418                                     Jefe de Planta   
13825  1112341619  24 feb - Sales Development Representative - On...   
13826  1112341705                        Animador HTML5 y Maquetador   
13827  1112341750  24 feb - Software Architects (USD$60K/yr) - On...   
13828  1112341763                                   Contador Publico   
13829  1112341811  24 Feb - Software Engineering Manager ($100K/y...   
13830  1112341494  Técnico electrónico - Turno tarde 12.30 a 22 HS   
13831  1112341585              Auxiliar de END para Oficina Técnica   
13832  1112341621  ESTUDIANTE DE ARQUITECTURA PARA ESTUDIO DE ARQ...   
13833  1112341402  24 Feb - VP of Inside Sales ($200K/yr) - Onlin...   
13834  1112341493  feb 24 - SVP of Outside Sales ($400K/yr) - Onl...   
13835  1112341524  24 Feb - VP of Technical Product Management - ...   
13836  1112341587  24 Feb - VP of Outside Sales ($200K/yr) - Onli...   
13837  1112341676  Feb 24 - VP of Technical Sales ($200K/yr) - On...   
13838  1112341708  Feb 24 - VP of Engineering ($200K/yr) - Online...   
13839  1112341940  24 Feb - Software Engineer ($30k/yr) - Online ...   
13840  1112341969  24 Feb - Financial Analysts - Online Hiring To...   
13841  1112342049  24 Feb - L2 Customer Support Architect (USD$60...   
13842  1112342073  24 Feb - VP of SaaSOps - Online